### PR TITLE
Update ComputerWorld.xml

### LINKED ISSUE #12313: Add rule for pcworld.com & computerworld.com
www.pcworld.com & www.computerworld.com now have a valid HTTPS certificate but are not HTTPS by default. I do recall there was a rulesets but may have been removed.

### PR DESCRIPTION
**Pull request by christianheinrichs**
Context: PR #12338 

Should close #12313.

Since the initial comments were pretty complex, I have left them mostly intact and moved my own research into a new XML comment block.

### CODE REVIEW

File: src/chrome/content/rules/ComputerWorld.xml [hunk 1]
```diff
@@ -1,85 +1,56 @@
...
 		- computerworld.idg.netdna-cdn.com
-
-			- -ssl doesn't exist
+			- SSL doesn't exist
 			- cdn[1-4].computerworlduk.com
 
-
 	Nonfunctional domains:
-
 		- blogs.computerworld.com *
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: WFM

File: src/chrome/content/rules/ComputerWorld.xml [hunk 2]
```diff
@@ -1,85 +1,56 @@
...
-		- resources.computerworld.com	(refused)
+		- resources.computerworld.com (refused)
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: WFM

File: src/chrome/content/rules/ComputerWorld.xml [hunk 3]
```diff
@@ -1,85 +1,56 @@
...
 		- (www.)computerworlduk.com *
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: WFM (both)

File: src/chrome/content/rules/ComputerWorld.xml [hunk 4]
```diff
@@ -1,85 +1,56 @@
...
 		- cdn[1-4].computerworlduk.com	(404; mismatched, CN: *.netdna-ssl.com)
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: Does not exist anymore

File: src/chrome/content/rules/ComputerWorld.xml [hunk 5]
```diff
@@ -1,85 +1,56 @@
...
-
-		- ^		(works; expired 2010-07-23, cert only matches www)
-		- events	(works; mismatched, CN: *.eiseverywhere.com)
-		- itjobs	(works; mismatched, CN: *.jobamatic.com)
-		- m		(shows mobify.com; mismatched, CN: *.mobify.com)
-
+		- ^	(works; expired 2010-07-23, cert only matches www)
+		- events (works; mismatched, CN: *.eiseverywhere.com)
+		- itjobs (works; mismatched, CN: *.jobamatic.com)
+		- m	(shows mobify.com; mismatched, CN: *.mobify.com)
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: WFM

File: src/chrome/content/rules/ComputerWorld.xml [hunk 6]
```diff
@@ -1,85 +1,56 @@
...
 
-
 	Problematic computerworld.com subdomains:
-
-		- ^		(works; expired 2010-07-23, cert only matches www)
-		- events	(works; mismatched, CN: *.eiseverywhere.com)
-		- itjobs	(works; mismatched, CN: *.jobamatic.com)
-		- m		(shows mobify.com; mismatched, CN: *.mobify.com)
-
+		- ^	(works; expired 2010-07-23, cert only matches www)
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: WFM

File: src/chrome/content/rules/ComputerWorld.xml [hunk 7]
```diff
@@ -1,85 +1,56 @@
...
+		- events (works; mismatched, CN: *.eiseverywhere.com)
+		- itjobs (works; mismatched, CN: *.jobamatic.com)
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: Does not exist anymore

File: src/chrome/content/rules/ComputerWorld.xml [hunk 8]
```diff
@@ -1,85 +1,56 @@
...
 
-
 	Fully covered domains:
-
 		- events.computerworld.com	(→ eiseverywhere.com)
-		- itjobs.compterworld.com	(→ computerworld.jobamatic.com)
-
+		- itjobs.computerworld.com	(→ computerworld.jobamatic.com)
 
 	Mixed content:
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: This block is outdated, you may remove it completely.

File: src/chrome/content/rules/ComputerWorld.xml [hunk 9]
```diff
@@ -1,85 +1,56 @@
...
 <!--
-Disabled by https-everywhere-checker because:
-Fetch error: http://computerworld.com/ => https://www.computerworld.com/: (51, "SSL: no alternative certificate subject name matches target host name 'www.computerworld.com'")
-
-Disabled by https-everywhere-checker because:
-Fetch error: http://computerworld.com/ => https://www.computerworld.com/: (28, 'Connection timed out after 10001 milliseconds')
 	For other IDG coverage, see IDG.se.xml.
 
-
 	CDN buckets:
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: This block is outdated, you may remove it completely.

File: src/chrome/content/rules/ComputerWorld.xml [hunk 10]
```diff
@@ -96,38 +67,56 @@ Fetch error: http://computerworld.com/ => https://www.computerworld.com/: (28, '
...
-
+<!--
+	Invalid security certificate:
+		- events.computerworld.com
+
+	Redirects:
+		- resources.computerworld.com → resources.computerworld.com/ccd/login
+
+	Server not found:
+		- itjobs.computerworld.com
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: You can remove non existent hosts.

File: src/chrome/content/rules/ComputerWorld.xml [hunk 11]
```diff
@@ -96,38 +67,56 @@ Fetch error: http://computerworld.com/ => https://www.computerworld.com/: (28, '
...
-
 -->
-<ruleset name="ComputerWorld (partial)" platform="mixedcontent" default_off='failed ruleset test'>
-
+<!--
+	Invalid security certificate:
+		- events.computerworld.com
+
+	Redirects:
+		- resources.computerworld.com → resources.computerworld.com/ccd/login
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: WFM over HTTPS

File: src/chrome/content/rules/ComputerWorld.xml [hunk 12]
```diff
@@ -96,38 +67,56 @@ Fetch error: http://computerworld.com/ => https://www.computerworld.com/: (28, '
 	** Unsecurable, but who cares. 404s anyway.
 	*** Unsecurable, doesn't trip MCB anyway
 
-
 		All secured by us, apart from:
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: This block is outdated, you may remove it completely.

File: src/chrome/content/rules/ComputerWorld.xml [hunk 13]
```diff
@@ -96,38 +67,56 @@ Fetch error: http://computerworld.com/ => https://www.computerworld.com/: (28, '
...
-
-	<!--securecookie host="^\.computerworld\.com$" name="^(idglg_ref_domain|mobify|s_pers|s_sess|__vr[flmuy]|__vrid)$" /-->
-	<securecookie host="^\.computerworld\.com$" name=".+" />
-
-
-	<rule from="^http://(?:www\.)?computerworld\.com/"
-		to="https://www.computerworld.com/"/>
-
-	<rule from="^http://events\.computerworld\.com/"
-		to="https://eiseverywhere.com/ehome/Computerworld/" />
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: We could keep that one

File: src/chrome/content/rules/ComputerWorld.xml [hunk 14]
```diff
@@ -96,38 +67,56 @@ Fetch error: http://computerworld.com/ => https://www.computerworld.com/: (28, '
...
-
-	<rule from="^http://itjobs\.computerworld\.com/"
-		to="https://computerworld.jobamatic.com/" />
+	<target host="www.computerworld.com" />
+	<target host="blogs.computerworld.com" />
+	<target host="m.computerworld.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]:
```
feeds.computerworld.com
itfinance.computerworld.com
itgovernment.computerworld.com
itreports.computerworld.com
masters.computerworld.com
mobile.computerworld.com
solutioncenters.computerworld.com

**Comment by J0WI** [comment]
Two comment blocks is a bad practice. Please merge/remove them.

**Comment by christianheinrichs** [comment]
@J0WI Thanks for the extensive code review on this one, learned quite a bit.

I have applied the requested changes in the second commit.

File: src/chrome/content/rules/ComputerWorld.xml (hunk 4 above)
Comment (christianheinrichs) [reply src/chrome/content/rules/ComputerWorld.xml]: @J0WI https://cdn1.computerworlduk.com, https://cdn2.computerworlduk.com, https://cdn3.computerworlduk.com and https://cdn4.computerworlduk.com WFM, so I moved them to the corresponding target tags.

File: src/chrome/content/rules/ComputerWorld.xml [hunk 15]
```diff
@@ -1,133 +1,67 @@
...
-
-
-	Nonfunctional domains:
-
-		- blogs.computerworld.com *
-		- resources.computerworld.com	(refused)
-		- (www.)computerworlduk.com *
-		- cdn[1-4].computerworlduk.com	(404; mismatched, CN: *.netdna-ssl.com)
-
 	* Dropped
```
Comment (J0WI) [inline src/chrome/content/rules/ComputerWorld.xml]: Please also remove this line

**Comment by J0WI** [comment]
LGTM, just one minor issue in the comments.

File: src/chrome/content/rules/ComputerWorld.xml (hunk 15 above)
Comment (christianheinrichs) [reply src/chrome/content/rules/ComputerWorld.xml]: Done.

**Comment by J0WI** [comment]
Thanks!

**Pull request merged**